server/pbenchinacan/deploy

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -e
+
+#
+# This script deploys a Pbench Server instance from a container image.
+#
+# Steps performed:
+#   - The Pbench Dashboard application files are copied from the build directory
+#     into the SRV_PBENCH file system.  (If the build directory does not exist,
+#     an empty directory is created at the input and output locations, a warning
+#     is printed, and the Pbench Server is deployed without a Dashboard.)
+#   - If present in the current directory, the Nginx favicon file is copied into
+#     the appropriate location in the SRV_PBENCH file system.
+#   - The ownership of the files in the SRV_PBENCH file system will are set for
+#     access inside the container.
+#   - The Pbench Server container is started and run as a detached process.
+#
+# The key inputs are the following environment variables:
+#   - PB_DEPLOY_FILES:  the location (on the host) of the configuration files
+#                       for the Pbench Server and rsyslog.
+#   - SRV_PBENCH:  the location of the file system (on the host) which will be
+#                       mounted inside the container at /srv/pbench.
+#
+# Other inputs which can be provided by environment variables include:
+#   - the name and pull policy for the Pbench Server container image
+#   - the location of the Pbench Dashboard app files and the Nginx Favicon
+#   - the location of the Pbench Git checkout (used to locate the above)
+#
+# In all cases, suitable defaults are provided for any value which is undefined.
+#
+
+#+
+# Configuration
+#-
+
+# Pbench Server container image
+${PB_SERVER_IMAGE:="images.paas.redhat.com/pbench/pbench-server:main"}
+${PB_SERVER_IMAGE_PULL_POLICY:="newer"}
+
+# Locations on the host
+${PB_DASHBOARD_DIR:=${GITTOP:=$(git rev-parse --show-toplevel)}/dashboard/build}
+${NGINX_FAVICON:=${GITTOP}/server/lib/config/color-square.ico}
+${PB_DEPLOY_FILES:=${HOME}/Deploy}
```

Comment:
I assume there's some significance here to defaulting to an odd directory outside the source tree and which we expect normally to override in `run-pbench-in-a-can`? If it can't be easily explained/justified here, perhaps it shouldn't be a default?

Reply:
Yeah:  this is "upward user compatibility" with the existing `deploy` script which Peter is already using for deploying the Staging server.  (In the description above, I promised a bunch of questions...this one is on that list.)